src/Optimizer/Transformer/MinifyHtml.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace AmpProject\Optimizer\Transformer;
+
+use AmpProject\Dom\Document;
+use AmpProject\Dom\Element;
+use AmpProject\Optimizer\Configuration\MinifyHtmlConfiguration;
+use AmpProject\Optimizer\ErrorCollection;
+use AmpProject\Optimizer\Transformer;
+use AmpProject\Optimizer\TransformerConfiguration;
+use DOMComment;
+use DOMNode;
+use DOMText;
+
+/**
+ * Transformer that that minifies HTML.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class MinifyHtml implements Transformer
+{
+
+    /**
+     * Configuration store to use.
+     *
+     * @var TransformerConfiguration
+     */
+    private $configuration;
+
+    /**
+     * Store nodes for later deletion
+     *
+     * @var array
+     */
+    private $nodesToRemove;
```

Comment:
These can better be managed via the return value of the different `minify*()` methods, which also makes semantic sense.
```suggestion
```